ocs_ci/deployment/aws.py

```diff
@@ -134,18 +152,247 @@ def destroy_cluster(self, log_level="DEBUG"):
+
+            # Change dir back to ocs-ci dir
+            os.chdir(cidir)
+
+            if proc.returncode:
+                logger.error(stderr)
+                raise exceptions.CommandFailed("upi install script failed")
+            logger.info(stdout)
+
+            self.test_cluster()
+
+    def deploy_ocp(self, log_cli_level='DEBUG'):
+        """
+        OCP deployment specific to AWS UPI
+
+        Args:
+             log_cli_level (str): openshift installer's log level
+                (default: 'DEBUG')
+        """
+        super(AWSUPI, self).deploy_ocp(log_cli_level)
+        volume_size = config.ENV_DATA.get('DEFAULT_EBS_VOLUME_SIZE', 100)
+        # existing function looks for terraform files
+        self.add_volume(volume_size)
+
+    def destroy_cluster(self, log_level="DEBUG"):
+        """
+        Destroy OCP cluster for AWS UPI
+
+        Args:
+            log_level (str): log level for openshift-installer (
+                default:DEBUG)
+        """
+        cluster_name = get_cluster_name(self.cluster_path)
+        # Destroy extra volumes
+        self.destroy_volumes()
+
+        # Create cloudformation client
+        cf = boto3.client('cloudformation')
+
+        # Delete master, bootstrap, security group, and worker stacks
+        suffixes = ['ma', 'bs', 'sg']
+        # TODO: read in num_workers in a better way
+        num_workers = int(os.environ.get('num_workers', 3))
+        for i in range(num_workers - 1, -1, -1):
+            suffixes.insert(0, f'no{i}')
+        stack_names = [f'{cluster_name}-{suffix}' for suffix in suffixes]
+        for stack_name in stack_names:
+            logger.info("Destroying stack: %s", stack_name)
+            cf.delete_stack(StackName=stack_name)
+            verify_stack_deleted(stack_name)
+
+        # Call openshift-installer destroy cluster
+        super(AWSUPI, self).destroy_cluster(log_level)
+
+        # Delete inf and vpc stacks
+        suffixes = ['inf', 'vpc']
+        stack_names = [f'{cluster_name}-{suffix}' for suffix in suffixes]
+        for stack_name in stack_names:
+            logger.info("Destroying stack: %s", stack_name)
+            cf.delete_stack(StackName=stack_name)
+            verify_stack_deleted(stack_name)
+
+        # Delete openshift-misc repository from cluster_path
+        openshift_misc_path = os.path.join(self.cluster_path, 'openshift-misc')
+        logger.info(
+            "Removing openshift-misc directory located at %s",
+            openshift_misc_path
+        )
+        shutil.rmtree(openshift_misc_path)
+
+
+def get_infra_id(cluster_path):
+    """
+    Get infraID from metadata.json in given cluster_path
+
+    Args:
+        cluster_path: path to cluster install directory
+
+    Returns:
+        str: metadata.json['infraID']
+
+    """
+    metadata_file = os.path.join(cluster_path, "metadata.json")
+    with open(metadata_file) as f:
+        metadata = json.loads(f.read())
+    return metadata.get("infraID")
```

Comment:
I would rather use direct access of the key with metadata['infraID'] and in case of some issue you will fail with the KeyError exception. If there is no such key you will return None which will lead to some confusion when debugging the issue.

Reply:
Using the `metadata` attribute directly was causing issues, and this method just works.

Reply:
Done